src/config/mongo.rs

```diff
@@ -0,0 +1,597 @@
+            connect_timeout: Default::default(),
+            credential: Default::default(),
+            direct_connection: Default::default(),
+            driver_info: Default::default(),
+            heartbeat_freq: Default::default(),
+            local_threshold: Default::default(),
+            max_idle_time: Default::default(),
+            max_pool_size: Default::default(),
+            min_pool_size: Default::default(),
+            read_concern: Default::default(),
+            repl_set_name: Default::default(),
+            retry_reads: Default::default(),
+            retry_writes: Default::default(),
+            selection_criteria: Default::default(),
+            server_api: Default::default(),
+            server_selection_timeout: Default::default(),
+            default_database: Default::default(),
+            tls: Default::default(),
+            write_concern: Default::default(),
+        }
+    }
+}
+
+impl Into<ClientOptions> for MongoConfig {
+    fn into(self) -> ClientOptions {
+        let builder = ClientOptions::builder()
+            .hosts(self.hosts.into_iter().map(Into::into).collect::<Vec<_>>())
+            .app_name(self.app_name)
+            .compressors(self.compressors.map(|v| v.into_iter().map(Into::into).collect()))
+            .connect_timeout(self.connect_timeout)
+            .credential(self.credential.map(Into::into))
+            .direct_connection(self.direct_connection)
+            .driver_info(self.driver_info.map(Into::into))
+            .heartbeat_freq(self.heartbeat_freq)
+            .local_threshold(self.local_threshold)
+            .max_idle_time(self.max_idle_time)
+            .max_pool_size(self.max_pool_size)
+            .min_pool_size(self.min_pool_size)
+            .read_concern(self.read_concern)
+            .repl_set_name(self.repl_set_name)
+            .retry_reads(self.retry_reads)
+            .retry_writes(self.retry_writes)
+            .selection_criteria(self.selection_criteria.map(Into::into))
+            .server_api(self.server_api)
+            .server_selection_timeout(self.server_selection_timeout)
+            .default_database(self.default_database)
+            .tls(self.tls.map(Into::into))
+            .write_concern(self.write_concern);
+        builder.build()
+    }
+}
+
+/// An enum representing the address of a MongoDB server.
+///
+/// Currently this just supports addresses that can be connected to over TCP, but alternative
+/// address types may be supported in the future (e.g. Unix Domain Socket paths).
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum ServerAddress {
+    /// A TCP/IP host and port combination.
+    Tcp {
+        /// The hostname or IP address where the MongoDB server can be found.
+        host: String,
+
+        /// The TCP port that the MongoDB server is listening on.
+        ///
+        /// The default is 27017.
+        port: Option<u16>,
```

Comment:
Should we use something like `std::net::SocketAddr` here?

Reply:
No, because SocketAddr has validation that we do not necessarily want, and also forces a port to exist. However, we could create some helpful trait impls to allow parsing from strings and such similar to SocketAddr.

Reply:
Ah, cool. If that's the case when can just keep it like this for now.